### DISCUSSION TITLE
split_bodies function for large amount of 3D-particles

### DISCUSSION THREAD

**Discussion by edouardbutaye** [pyvista/pyvista#5753] [Q&A] [answered]
### First check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the PyVista documentation, with the integrated search.
- [X] I already searched in Google "How to X in PyVista" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I will NOT make duplicate discussions or issues in the future.

### Commit to Help

- [X] I commit to help with one of those options.

### Sample Code What is the problem, question, or error?
Write a short description telling me what you are doing, what you expect to happen, and what is currently happening.

```python
import pyvista as pv
mesh=pv.read("champs_lagrangiens_0000.vtk") # read a vtk mesh containing 2134 spherical particles
particles=mesh.split_bodies(progress_bar=True) # split mesh into 2134 mesh: one for each particle
```

### Description

Hello,
I'm a PhD student working on simulation of fluid-particles flow using TrioCFD software. The interface between the fluid and particles is computed with the Front-Tracking method. The interface is therefore a moving Lagrangian mesh. The simulation returns the pressure and velocities components onto an Eulerian mesh as well as the coordinates of the Lagrangian mesh onto the Eulerian mesh.  I want to compute the hydrodynamic force exerted by the fluid over the particle's surface.

This requires to:
1. Split the lagrangian mesh to know the lagrangian mesh of each particle.
2. Find the eulerian cell that include a point (x,y,z).
3.  Interpolate eulerian velocity and pressure at random points included in the domain.

I think I can manage to code points 2 and 3. However, the first point is extremely slow using the "split_bodies" function. I have estimated the computation time to 10 hours. As I have to compute it for thousands of time steps, I cannot use this function.
Is there a fastest way to do it ?

### System Information

```shell
There is no log to show.
```

**Comment by darikg** [comment, marked as answer]
Is it any better to just use [the connectivity filter](https://docs.pyvista.org/version/stable/api/core/_autosummary/pyvista.datasetfilters.connectivity) to label different components, and operating on the original mesh using the 'RegionId' indices?

**Reply by edouardbutaye** [reply]
Hi,
Thanks a lot for your answer. It is a very good advice, I'm quite new to pyvista and I guess I missed this keyword during the search. According to the documentation, it seems quite appropriate.
